ntoskrnl/io/iomgr/deviface.c

```diff
@@ -275,36 +276,89 @@ IoOpenDeviceInterfaceRegistryKey(IN PUNICODE_STRING SymbolicLinkName,
     return Status;
 }
 
-/*++
- * @name IoGetDeviceInterfaceAlias
- * @unimplemented
- *
- * Returns the alias device interface of the specified device interface
- * instance, if the alias exists.
- * Documented in WDK.
- *
- * @param SymbolicLinkName
- *        Pointer to a string which identifies the device interface instance
- *
- * @param AliasInterfaceClassGuid
- *        See WDK
- *
- * @param AliasSymbolicLinkName
- *        See WDK
- *
- * @return Three different NTSTATUS values in case of errors, and STATUS_SUCCESS
- *         otherwise (see WDK for details)
- *
- * @remarks Must be called at IRQL = PASSIVE_LEVEL in the context of a system thread
- *
- *--*/
+static
 NTSTATUS
-NTAPI
-IoGetDeviceInterfaceAlias(IN PUNICODE_STRING SymbolicLinkName,
-                          IN CONST GUID *AliasInterfaceClassGuid,
-                          OUT PUNICODE_STRING AliasSymbolicLinkName)
+IopSeparateSymbolicLink(IN PUNICODE_STRING SymbolicLinkName,
+                        OUT PUNICODE_STRING DeviceString,
+                        OUT PUNICODE_STRING GuidString,
+                        OUT PUNICODE_STRING ReferenceString)
 {
-    return STATUS_NOT_IMPLEMENTED;
+    UNICODE_STRING DosDevicesPrefix1 = RTL_CONSTANT_STRING(L"\\??\\");
+    UNICODE_STRING DosDevicesPrefix2 = RTL_CONSTANT_STRING(L"\\\\?\\");
+    UNICODE_STRING LinkNameNoPrefix;
+    UNICODE_STRING DeviceStringRealP, GuidStringRealP, ReferenceStringRealP;
```

Comment:
What does the 'P' suffix mean?

Reply:
It means that the variable is private. I added this to avoid confusing with the names of the function parameters, whose are named very similarly.
Also this suffix is used in some internal functions above btw (see `OpenRegistryHandlesFromSymbolicLink`). This one is also internal, so I just used general coding style.

Reply:
"variable is private" --> that doesn't really mean anything here in this context: All the variables local to a function are "private" in that sense...

Reply:
OK, then the 1st suffix is enough.